src/preload/test/ext/requests.js

```diff
@@ -0,0 +1,148 @@
+describe('preload extension sends requests which match element requests', function() {
+    let requests = [];
+  
+    beforeEach(function() {
+      this.server = makeServer()
+      const defaultOnCreateFunction = this.server.xhr.onCreate
+      this.server.xhr.onCreate = function(xhr) {
+        defaultOnCreateFunction(xhr)
+        requests.push(xhr)
+      }
+      this.clock = sinon.useFakeTimers()
+      clearWorkArea()
+      getWorkArea().setAttribute('hx-ext', 'preload')
+    })
+    
+    afterEach(function() {
+      this.server.restore()
+      requests = []
+      this.clock.restore()
+      getWorkArea().removeAttribute('hx-ext')
+      clearWorkArea()
+    })
+
+    it('includes HX-Preloaded header when preloading hyperlink', function() {
+      const hyperlink = make('<a href="/test" preload>Link</a>')
+  
+      htmx.trigger(hyperlink, 'mousedown')
+      
+      should.equal(requests.length, 1)
+      requests[0].requestHeaders.should.deep.contain({"HX-Preloaded": "true"})
+    })
+
+    it('includes HX-Preloaded header when preloading xh-boosted hyperlink', function() {
+      const hyperlink = make('<a href="/test" hx-boost="true" preload>Link</a>')
+  
+      htmx.trigger(hyperlink, 'mousedown')
+      
+      should.equal(requests.length, 1)
+      requests[0].requestHeaders.should.deep.contain({"HX-Preloaded": "true"})
+    })
+
+    it('includes HX-Preloaded header when preloading button with hx-get attribute', function() {
+      const button = make('<button hx-get="/test" preload>Button</button>')
+
+      htmx.trigger(button, 'mousedown')
+  
+      should.equal(requests.length, 1)
+      requests[0].requestHeaders.should.deep.contain({"HX-Preloaded": "true"})
+    })
+
+    it('matches url and includes headers for hx-boosted hyperlink', function() {
+      const hyperlink = make('<a href="/test" hx-boost="true" preload>Link</a>')
+  
+      htmx.trigger(hyperlink, 'mousedown')
+      this.server.respond()
+      hyperlink.click()
+      
+      should.equal(requests.length, 2)
+      should.equal(requests[0].url, requests[1].url)
+      should.equal(requests[0].method, requests[1].method)
+      requests[0].requestHeaders.should.deep.contain(requests[1].requestHeaders)
```

Comment:
Maybe we should also check, on top of the preloaded request containing all the headers of the htmx click request, that the click request does **not** include the preloaded header? Just to ensure we don't mess this up in the future if we were to make new changes to this extension over time

Reply:
Here you go https://github.com/bigskysoftware/htmx-extensions/pull/106/commits/2f105eff533e2e2099c94d7249113f0e5557fdb2 😉